src/pages/workspaces/workspace/analysis/runtime-common.js

```diff
@@ -92,8 +92,9 @@ export const RuntimeStatusMonitor = ({ runtime, onRuntimeStoppedRunning = _.noop
 }
 
 export const AuthenticatedCookieSetter = () => {
-  const { registrationStatus } = useStore(authStore)
-  return registrationStatus === userStatus.registeredWithTos && getLocalPref(cookiesAcceptedKey) !== false ? h(PeriodicCookieSetter) : null
+  const { termsOfService } = useStore(authStore)
+  return (termsOfService.userAcceptedTos || termsOfService.isGracePeriodEnabled) &&
```

Comment:
Could this use `termsOfService.userNeedsToAcceptTos` so that the logic is centralized?

https://github.com/DataBiosphere/terra-ui/blob/4f7d70b8eb99326e59f1a8d8c76e57f2c9ab5421/src/libs/auth.js#L342-L343

Reply:
I did that originally, but I like this way better to insulate against changes to the logic behind `userNeedsToAcceptTos`. This set of flags also better describes what exactly we are checking

Reply:
I disagree. That's exactly the sort of logic we'd want to centralize and not have to re-implement in multiple places.

Reply:
Oops, I was wrong. Looks like I was misinterpreting `userNeedsToAcceptTos` and that the way you have it is the best option with the current values in `authStore`. Looks like #3671 will make this easier.